bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/page/PageMetadataImpl.java

```diff
@@ -0,0 +1,101 @@
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+package com.adobe.cq.commerce.core.components.internal.models.v1.page;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.ScriptVariable;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+
+import com.adobe.cq.commerce.core.components.models.page.PageMetadata;
+import com.adobe.cq.commerce.core.components.models.product.Product;
+import com.adobe.cq.commerce.core.components.models.productlist.ProductList;
+import com.adobe.cq.commerce.core.components.utils.SiteNavigation;
+import com.day.cq.commons.jcr.JcrConstants;
+import com.day.cq.wcm.api.Page;
+
+@Model(
+    adaptables = SlingHttpServletRequest.class,
+    adapters = PageMetadata.class)
+public class PageMetadataImpl implements PageMetadata {
+
+    @Self
+    private SlingHttpServletRequest request;
+
+    @ScriptVariable
+    private Page currentPage;
+
+    @ScriptVariable
+    private ValueMap properties;
+
+    private PageMetadata provider;
+
+    @PostConstruct
+    void initModel() {
+        if (isProductPage()) {
+            Product product = request.adaptTo(Product.class);
+            provider = product;
+        } else if (isCategoryPage()) {
+            ProductList productList = request.adaptTo(ProductList.class);
+            provider = productList;
+        }
+    }
+
+    @Override
+    public String getMetaDescription() {
+        String metaDescription = provider != null ? provider.getMetaDescription() : null;
+        return metaDescription != null ? metaDescription : properties.get(JcrConstants.JCR_DESCRIPTION, String.class);
+    }
+
+    @Override
+    public String getMetaKeywords() {
+        String metaKeywords = provider != null ? provider.getMetaKeywords() : null;
+        if (metaKeywords == null && currentPage instanceof com.adobe.cq.wcm.core.components.models.Page) {
+            metaKeywords = ((com.adobe.cq.wcm.core.components.models.Page) currentPage).getKeywords().toString();
+        }
+        return metaKeywords;
+    }
+
+    @Override
+    public String getMetaTitle() {
+        String metaTitle = provider != null ? provider.getMetaTitle() : null;
+        return metaTitle != null ? metaTitle : currentPage.getTitle();
```

Comment:
Here you return the meta title if available, but otherwise the product name:
https://github.com/adobe/aem-core-cif-components/blob/973747b528bbae6fad5e6e2bda0ef1a2fa48014e/bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/product/ProductImpl.java#L405-L408

So the page title is actually never used. Do we really want that? I would assume we go for the meta title if available and otherwise some kind of concatenation of page title and product name. Maybe we should check back with our PMs :)

Reply:
I think it is ok. For PDP/PLP we agreed that commerce is the source. The page title will still be used for all other pages using the page component.

Reply:
Exactly, this is what we discussed/agreed. Customers would have to either override the model and these methods if they want to change this, and they can also override the `head.html` file and for example add a suffix to all pages with the sites name, to have something like `Chaz Hoodie - Venia Store`.